include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -251,6 +251,110 @@ struct disjoint_segment_box
     }
 };
 
+
+struct disjoint_segment_box_with_info_compute
+{
+    template <typename Segment, typename Box, typename Strategy, typename Result>
+    static inline void apply(Segment const& segment,
+                             Box const& box,
+                             Strategy const& strategy,
+                             Result& result)
```

Comment:
Interesting, so here the `Result` is taken as an output argument instead of returned from function. Is there a reason why isn't one pattern consistently applied in all places?

Reply:
Btw, if it was implemented like this in all places then the point type could be taken from the Segment type passed by the user at the highest level. But then I guess it wouldn't be possible to use existing `distance` algorithm to return the info and you'd have to implement yet another `_with_info` algorithm. Correct?

Reply:
The main function here is apply of `disjoint_segment_box_with_info`. In the `_compute` helper I used the output argument just because I found it simpler to derive the type of the result and since it is only called from `disjoint_segment_box_with_info`